applications/adv_networking_bench/cpp/rivermax_bench_op_tx.h

```diff
@@ -0,0 +1,644 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "advanced_network/common.h"
+#include "advanced_network/kernels.h"
+#include "kernels.cuh"
+#include "holoscan/holoscan.hpp"
+#include <queue>
+#include <arpa/inet.h>
+#include <assert.h>
+#include <sys/time.h>
+
+using namespace holoscan::advanced_network;
+
+namespace holoscan::ops {
+/**
+ * @brief: Holds the essential frame data.
+ */
+class FrameBuffer {
+ public:
+  /**
+   * @brief: Constructor that allocates internal memory.
+   *
+   * @param [in] buffer_size: Size of the buffer to allocate.
+   */
+  explicit FrameBuffer(size_t buffer_size)
+      : buffer_(new uint8_t[buffer_size]), buffer_size_(buffer_size) {}
+
+  /**
+   * @brief: Move constructor.
+   */
+  FrameBuffer(FrameBuffer&& other) noexcept
+      : buffer_(std::move(other.buffer_)), buffer_size_(other.buffer_size_) {}
+
+  /**
+   * @brief: Move assignment operator.
+   */
+  FrameBuffer& operator=(FrameBuffer&& other) noexcept {
+    if (this != &other) {
+      buffer_ = std::move(other.buffer_);
+      buffer_size_ = other.buffer_size_;
+    }
+    return *this;
+  }
+
+  // Delete copy constructor and assignment operator
+  FrameBuffer(const FrameBuffer&) = delete;
+  FrameBuffer& operator=(const FrameBuffer&) = delete;
+  ~FrameBuffer() = default;
+
+  /**
+   * @brief: Return pointer to the buffer.
+   *
+   * @return: Pointer to the buffer.
+   */
+  uint8_t* get() const {
+    return buffer_.get();
+  }
+
+  /**
+   * @brief: Return size of the buffer.
+   *
+   * @return: Size of the buffer.
+   */
+  size_t get_size() const {
+    return buffer_size_;
+  }
+
+ private:
+  /* Smart pointer for the owned memory */
+  std::unique_ptr<uint8_t[]> buffer_;
+  /* Size of the buffer */
+  size_t buffer_size_;
+};
+
+/**
+ * @brief Thread-safe buffer for concurrent item production and consumption.
```

Comment:
Any reason you don't want to use a DPDK ring for this?

Reply:
@cliffburdick can you capture the ring suggestion in a JIRA ticket to address later? maybe you already have.